crates/rspack_plugin_javascript/src/parser_plugin/common_js_imports_parse_plugin.rs

```diff
@@ -47,23 +49,107 @@ fn create_commonjs_require_context_dependency(
   CommonJsRequireContextDependency::new(options, span.into(), (start, end), parser.in_try)
 }
 
+fn create_require_resolve_context_dependency(
+  parser: &mut JavascriptParser,
+  param: &BasicEvaluatedExpression,
+  expr: &Expr,
+  range: RealDependencyLocation,
+  weak: bool,
+) -> RequireResolveContextDependency {
+  let start = range.start;
+  let end = range.end;
+  let result = create_context_dependency(param, expr, parser);
+  let options = ContextOptions {
+    mode: if weak {
+      ContextMode::Weak
+    } else {
+      ContextMode::Sync
+    },
+    recursive: true,
+    reg_exp: context_reg_exp(&result.reg, "", None, parser),
+    include: None,
+    exclude: None,
+    category: DependencyCategory::CommonJS,
+    request: format!("{}{}{}", result.context, result.query, result.fragment),
+    context: result.context,
+    namespace_object: ContextNameSpaceObject::Unset,
+    group_options: None,
+    replaces: result.replaces,
+    start,
+    end,
+    referenced_exports: None,
+    attributes: None,
+  };
+  RequireResolveContextDependency::new(options, range, parser.in_try)
+}
+
 pub struct CommonJsImportsParserPlugin;
 
 impl CommonJsImportsParserPlugin {
-  fn add_require_resolve(&self, parser: &mut JavascriptParser, node: &CallExpr, weak: bool) {
-    if !node.args.is_empty()
-      && let Some(Lit::Str(str)) = node.args.first().and_then(|x| x.expr.as_lit())
-    {
-      let range: RealDependencyLocation = node.span.into();
+  fn process_resolve(&self, parser: &mut JavascriptParser, call_expr: &CallExpr, weak: bool) {
+    if call_expr.args.len() != 1 {
+      return;
+    }
+
+    let argument_expr = &call_expr.args[0].expr;
+    let param = parser.evaluate_expression(argument_expr);
+    let require_resolve_header_dependency = Box::new(RequireResolveHeaderDependency::new(
+      call_expr.callee.span().into(),
+    ));
+
+    if param.is_conditional() {
+      for option in param.options() {
+        if !self.process_resolve_item(parser, option, weak) {
+          self.process_resolve_context(parser, option, argument_expr, weak);
+        }
+      }
+      parser.dependencies.push(require_resolve_header_dependency);
+    } else {
+      if !self.process_resolve_item(parser, &param, weak) {
+        self.process_resolve_context(parser, &param, argument_expr, weak);
+      }
+      parser.dependencies.push(require_resolve_header_dependency);
+    }
+  }
+
+  fn process_resolve_item(
+    &self,
+    parser: &mut JavascriptParser,
+    param: &BasicEvaluatedExpression,
+    weak: bool,
+  ) -> bool {
+    if param.is_string() {
       parser
         .dependencies
         .push(Box::new(RequireResolveDependency::new(
-          str.value.to_string(),
-          range,
+          param.string().to_string(),
+          param.range().into(),
```

Comment:
I see.. I recommend moving -1 here, there are some mess about BasicEvaluatedExpression.range, we will refactor it, moving -1 here should easy for refactoring in the future

Reply:
Yes, it took me some time to realize that BasicEvaluatedExpression.range might not be entirely accurate.